### ISSUE TITLE
Convert `expression has unnecessary parentheses` warnings into linter inspection

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#9197]
## Environment

* **IntelliJ Rust plugin version:** 0.4.176.4797-222-nightly
* **Rust toolchain version:** 1.63.0 (4b91a6ea7 2022-08-08) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.2 Ultimate Edition (IU-222.3345.118)
* **Operating system:** macOS 12.3.1
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
Currently, the plugin detects unnecessary parens is some language constructions like `if` condition, expression in `for` loop, etc.

But the current implementation can be improved in some places:
- it should be possible to turn it off
- take into account `unused_parens` and `unused` [lints](), i.e. the plugin shouldn't show warnings if there is `#[allow(unused_parens)]`
- provide suppress quick fixes to insert `#[allow(unused_parens)]`
- messages should be the same as compiler produces

<img width="601" alt="image" src="https://user-images.githubusercontent.com/2539310/184639932-addf6457-377a-4206-b15a-40ea33643c13.png">

From the technical point of view, all these things (except messages) can be achieved via conversion of [this part](https://github.com/intellij-rust/intellij-rust/blob/4885ab4e9f02645a51e6020bdb4d60d9752b216d/src/main/kotlin/org/rust/ide/annotator/RsExpressionAnnotator.kt#L81-L118) of [RsExpressionAnnotator](https://github.com/intellij-rust/intellij-rust/blob/4885ab4e9f02645a51e6020bdb4d60d9752b216d/src/main/kotlin/org/rust/ide/annotator/RsExpressionAnnotator.kt#L22) into [lint inspection](https://github.com/intellij-rust/intellij-rust/blob/4885ab4e9f02645a51e6020bdb4d60d9752b216d/src/main/kotlin/org/rust/ide/inspections/lints/RsLintInspection.kt#L22)

## Steps to reproduce
```rust
fn main() {
    for i in (1..10) {}
    if (1 > 2) {
        println!("Hello!");
    }
}

```
